pkg/server/singleprocess/service_runner.go

```diff
@@ -82,15 +87,19 @@ func (s *Service) AdoptRunner(
 	ctx context.Context,
 	req *pb.AdoptRunnerRequest,
 ) (*empty.Empty, error) {
-	if err := serverptypes.ValidateAdoptRunnerRequest(req); err != nil {
+	var err error
+	if err = serverptypes.ValidateAdoptRunnerRequest(req); err != nil {
 		return nil, err
 	}
-
-	var err error
+	log := hclog.FromContext(ctx)
 	if req.Adopt {
-		err = s.state(ctx).RunnerAdopt(req.RunnerId, false)
+		if err = s.state(ctx).RunnerAdopt(req.RunnerId, false); err != nil {
+			return &empty.Empty{}, hcerr.Externalize(log, err, "failed to adopt runner", "id", req.RunnerId)
+		}
 	} else {
-		err = s.state(ctx).RunnerReject(req.RunnerId)
+		if err = s.state(ctx).RunnerReject(req.RunnerId); err != nil {
+			return &empty.Empty{}, hcerr.Externalize(log, err, "failed to reject runner", "id", req.RunnerId)
+		}
 	}
 
 	return &empty.Empty{}, err
```

Comment:
Totally unrelated to you, just a thing I noticed. Err has to be nil here, so it'd be nice to be explicit about that.
```suggestion
	return &empty.Empty{}, nil
```

Reply:
I did see this, but thought it was fine since default value for err would be nil anyways. I can change it to be explicit if that is preferred.